crates/opencascade/src/primitives.rs

```diff
@@ -219,21 +268,42 @@ impl Wire {
     }
 
     pub fn fillet(&mut self, radius: f64) {
+        // Create a face from this wire
+        let mut face = Face::from_wire(self);
+        face.fillet(radius);
+        let wire = outer_wire(&face.inner);
+
+        self.inner = wire;
+    }
+
+    /// Chamfer the wire by a given distance and angle in radians
+    pub fn chamfer_angle(&mut self, distance: f64, angle: Angle) {
         // Create a face from this wire
         let face = Face::from_wire(self);
-        // use BRepFilletAPI_MakeFillet2d
         let mut make_fillet = BRepFilletAPI_MakeFillet2d_ctor(&face.inner);
 
         // add all vertices from the face
         let face_shape = cast_face_to_shape(&face.inner);
 
         // We use a shape map here to avoid duplicates.
-        let mut shape_map = new_indexed_map_of_shape();
-        map_shapes(face_shape, TopAbs_ShapeEnum::TopAbs_VERTEX, shape_map.pin_mut());
-
-        for i in 1..=shape_map.Extent() {
-            let vertex = TopoDS_cast_to_vertex(shape_map.FindKey(i));
-            BRepFilletAPI_MakeFillet2d_add_fillet(make_fillet.pin_mut(), vertex, radius);
+        let mut vector_map = new_indexed_map_of_shape();
+        map_shapes(face_shape, TopAbs_ShapeEnum::TopAbs_VERTEX, vector_map.pin_mut());
+
+        let mut edge_map = new_indexed_map_of_shape();
+        map_shapes(face_shape, TopAbs_ShapeEnum::TopAbs_EDGE, edge_map.pin_mut());
+
+        // walk vertices - will chamfer edges connected to the vertex
+        // TODO should be a dict lookup of vertex to edge? Only works if both equal length - a enclosed shape
```

Comment:
Yep, I think the map is only removing duplicates, we will likely have to query for connectivity information from opencascade.